vars/cloudformation.groovy

```diff
@@ -199,6 +199,11 @@ def queryStackOutput(cf, config){
 def queryStackExport(cf, config){
   def result = cf.listExports(new ListExportsRequest())
   def export = result.getExports().find { it.getName().equals(config.query.toString()) }
+  
+  while(result.nextToken != null && export == null) {
```

Comment:
this could get stuck in an infinite loop if the export didn't exists as both conditions have to be met to exit. I would get all the results first and then search for the export.

checkout this example here https://github.com/base2Services/ciinabox-pipelines/blob/ead8d5871faf3bebc6639a618a165cedfcf595c7/vars/cloudformation.groovy#L609-L629